luaui/Widgets/map_start_position_suggestions.lua

```diff
@@ -0,0 +1,744 @@
+function widget:GetInfo()
+	return {
+		name = "Start Position Suggestions",
+		desc = "Show expert recommended starting locations on the map",
+		license = "GNU GPL, v2 or later",
+		layer = 10000,
+		enabled = true
+	}
+end
+
+--todo: gl4 (also make circles more transparent as you zoom in)
+
+local base64 = VFS.Include("common/luaUtilities/base64.lua")
+
+-- config
+-- ======
+
+local config = {
+	-- saved
+	hasRunBefore = false,
+
+	-- local
+	circleRadius = 300,
+	circleThickness = 6,
+
+	playerTextSize = 180,
+	roleTextSize = 100,
+	tutorialTextSize = 20,
+
+	spawnPointCircleColor = { 1, 1, 1, 0.6 },
+	baseCenterCircleColor = { 1, 1, 1, 0.4 },
+	usePlayerColorForSpawnPointCircle = true,
+
+	playerTextColor = { 1, 1, 1, 0.6 },
+	roleTextColor = { 1, 1, 1, 0.6 },
+
+	glowRadiusCoefficient = -0.3,
+
+	tooltipDelay = 0.37,
+	tooltipMaxWidthChars = 60,
+
+	hideTooltipsAfterBuild = true,
+
+	tutorialMaxWidthChars = 100,
+}
+
+-- engine call optimizations
+-- =========================
+
+local SpringGetCameraState = Spring.GetCameraState
+local SpringIsGUIHidden = Spring.IsGUIHidden
+local SpringGetCameraRotation = Spring.GetCameraRotation
+local SpringGetGameFrame = Spring.GetGameFrame
+
+local glColor = gl.Color
+local glDepthTest = gl.DepthTest
+local glPushMatrix = gl.PushMatrix
+local glTranslate = gl.Translate
+local glRotate = gl.Rotate
+local glPopMatrix = gl.PopMatrix
+
+-- types
+-- =====
+
+---@alias Timer userdata
+---
+---@alias StartPositionID string
+
+---@class ModOptionPosition
+---@field x number
+---@field y number
+
+---@class ModOptionTeamStartPositionEntry
+---@field spawnPoint StartPositionID
+---@field baseCenter StartPositionID
+---@field role string
+
+---@class ModOptionTeamStartPositions
+---@field starts ModOptionTeamStartPositionEntry[]
+---
+---@class ModOptionTeams
+---@field playersPerTeam number
+---@field teamCount number
+---@field sides ModOptionTeamStartPositions[]
+
+---@alias ModOptionPositions table<StartPositionID, ModOptionPosition>
+
+---@class ModOptionData
+---@field positions ModOptionPositions
+---@field team ModOptionTeams[]
+
+---@alias AllyTeamID number
+
+---@class WidgetMapPosition
+---@field x number
+---@field z number
+
+---@class WidgetTeamStartPositionEntry
+---@field spawnPoint WidgetMapPosition
+---@field baseCenter WidgetMapPosition?
+---@field role string
+
+---@alias WidgetStartPositions table<AllyTeamID, WidgetTeamStartPositionEntry[]>
+
+
+-- loading and processing code
+-- ===========================
+
+local function convertXYToXZ(p)
```

Comment:
Perhaps the input should just be XZ form the start?

Reply:
Yeah... In maps metadata I was already using the XY because it's more natural for coordinates on the plane so it's also what I used for start pos, and here, modoption is ~raw export from maps metadata without any conversions. I could do conversion during export if that's required, but per chat it's not required so I'm gonna resolve this thread.